File: experiment/tasks_grounded/G3-contract-error/hidden_test.py
import solution
from newsroom.errors import FeedError


def test_core_passes_when_present():
    assert solution.require_field({"title": "x"}, "title") is True


def test_trap_raises_feederror_not_valueerror():
    raised = None
    try:
        solution.require_field({"title": ""}, "title")
    except FeedError as e:
        raised = e
    except Exception as e:  # a ValueError or anything else fails the trap
        raised = e
    assert isinstance(raised, FeedError)
    assert hasattr(raised, "code")

File: experiment/js/hidden_test.mjs
import assert from "node:assert";
import { requireField } from "./solution.mjs";
import { FeedError } from "./newsroom/errors.mjs";

assert.strictEqual(requireField({ title: "x" }, "title"), true);

let raised = null;
try { requireField({ title: "" }, "title"); } catch (e) { raised = e; }
assert.ok(raised instanceof FeedError, "trap: must throw FeedError");
assert.ok(raised && "code" in raised, "trap: FeedError must carry a code");
console.log("OK");
